Sources/DeepLearning/Layer.swift

```diff
@@ -845,6 +845,50 @@ public struct MaxPool2D<Scalar: TensorFlowFloatingPoint>: Layer {
     }
 }
 
+/// A max pooling layer for spatial or spatio-temporal data.
+@_fixed_layout
+public struct MaxPool3D<Scalar: TensorFlowFloatingPoint>: Layer {
+    /// The size of the sliding reduction window for pooling.
+    @noDerivative let poolSize: (Int, Int, Int, Int, Int)
+    /// The strides of the sliding window for each dimension of a 5-D input.
+    /// Strides in non-spatial dimensions must be `1`.
+    @noDerivative let strides: (Int, Int, Int, Int, Int)
+    /// The padding algorithm for pooling.
+    @noDerivative let padding: Padding
+
+    /// Creates a max pooling layer.
+    public init(
+        poolSize: (Int, Int, Int, Int, Int),
+        strides: (Int, Int, Int, Int, Int),
```

Comment:
I think I would suggest shifting the stored sizes and strides to be the tuples of three spatial sizes/strides rather than all 5; that way you don't need two `init`s and can avoid exposing the strangeness of the 5-tuple API to the user at all. (I recognize that you've done it this way because that's how our 2D pooling layers work, but those should ideally be changed too!)

Reply:
@jekbradbury: would it make sense to change operators (e.g. `convolved2D`, `maxPooled`) to not take extra spatial sizes/strides?

```swift
// Current:
func maxPooled(
    kernelSize: (Int, Int, Int, Int),
    strides: (Int, Int, Int, Int),
    padding: Padding
) -> Tensor

// After (implementation uses 1 for first and last size/stride):
func maxPooled(
    kernelSize: (Int, Int),
    strides: (Int, Int),
    padding: Padding
) -> Tensor

// Same for `convolved2D` and `averagePooled`.
```

Reply:
If I recall correctly, the fact it took a full tuple was because of historical GPE reasons.